src/main.hpp

```diff
@@ -5,3 +5,7 @@
 #include "io/iomanager.hpp"
 
 std::shared_ptr<dmxfish::io::IOManager> get_iomanager_instance();
+
+void construct_iomanager();
+
+void destruct_iomanager();
```

Comment:
Please add the warnings that they should be only called from main or unit tests here as well.